src/utils.rs

```
@@ -164,40 +153,76 @@ pub fn kelvin_source_to_nonce<H: ByteHash>(source: &mut Source<H>) -> io::Result
     source.read_exact(&mut n).map(|_| Nonce(n))
 }
 
-/// Convert a [`JubJubProjective`] into affine and then serialize the `x` coordinate
-pub fn serialize_compressed_jubjub(p: &JubJubProjective, bytes: &mut [u8]) -> Result<usize, Error> {
-    p.into_affine().serialize(&[], bytes)?;
+/// Convert a [`JubJubExtended`] into affine and then serialize the `x` coordinate
+pub fn serialize_compressed_jubjub(p: &JubJubExtended, bytes: &mut [u8]) -> Result<usize, Error> {
+    let b = JubJubAffine::from(p).to_bytes();
+    bytes.copy_from_slice(&b[..]);
 
     Ok(COMPRESSED_JUBJUB_SERIALIZED_SIZE)
 }
 
-/// Deserialize a [`JubJubAffine`] from a slice of bytes, and convert it to [`JubJubProjective`]
-pub fn deserialize_compressed_jubjub(bytes: &[u8]) -> Result<JubJubProjective, Error> {
-    Ok(JubJubAffine::deserialize(bytes, &mut [])?.into_projective())
+/// Deserialize a [`JubJubAffine`] from a slice of bytes, and convert it to [`JubJubExtended`]
+pub fn deserialize_compressed_jubjub(bytes: &[u8]) -> Result<JubJubExtended, Error> {
+    if bytes.len() < 32 {
+        return Err(Error::InvalidParameters);
+    }
+
+    let mut array = [0u8; 32];
+    array.copy_from_slice(&bytes[..32]);
+    let result = JubJubAffine::from_bytes(array);
+    if result.is_none().unwrap_u8() == 1 {
+        return Err(Error::InvalidParameters);
+    }
+
+    Ok(JubJubExtended::from(result.unwrap()))
 }
```

This one makes sense to me. Since we want to return a propper error instead of a `CtOption`